Automated cherry pick of #105273: tests: Wait for pod collection to enter a Running state

claudiubelu opened the pull request:
Cherry pick of #105273 on release-1.21.

#105273: tests: Wait for pod collection to enter a Running state


#### What type of PR is this?

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

/kind flake
/sig testing
/sig windows

/priority important-soon

#### What this PR does / why we need it:

While running tests in parallel, especially those with higher loads than others, it might take some time for Pods to be Running, even more so if the image has to be pulled as well.

The test ``[sig-node] Pods should delete a collection of pods [Conformance]`` only waits for the for the pods to be scheduled before deleting them, and expects them to be gone in 1 minute, which can flake because of the above reasons. Note that the operations are in order, and kubelet runs them in order, which means that the pod first has to enter the Running state before attempting to delete it.

This commit waits for the Pods to enter the Running state first before deleting the entire collection.

Co-Authored-By: Antonio Ojea <aojea@redhat.com>

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #103671

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
NONE
```

claudiubelu committed:
tests: Wait for pod collection to enter a Running state

While running tests in parallel, especially those with higher loads
than others, it might take some time for Pods to be Running, even more
so if the image has to be pulled as well.

The test [sig-node] Pods should delete a collection of pods [Conformance]
only waits for the for the pods to be scheduled before deleting them, and
expects them to be gone in 1 minute, which can flake because of the above
reasons. Note that the operations are in order, and kubelet runs them in
order, which means that the pod first has to enter the Running state
before attempting to delete it.

This commit waits for the Pods to enter the Running state first before
deleting the entire collection.

Co-Authored-By: Antonio Ojea <aojea@redhat.com>

claudiubelu commented:
/kind flake
/priority important-soon

/sig windows
/sig testing

SergeyKanzhelev commented:
@claudiubelu is this test running for 1.21? Can you please give a testgrid link?

claudiubelu commented:
> 
> 
> @claudiubelu is this test running for 1.21? Can you please give a testgrid link?

Sure. Here is one job it runs in: https://testgrid.k8s.io/sig-windows-signal#aks-engine-windows-containerd-1.21

SergeyKanzhelev commented:
I don't see this test failing in the linked grid. It is not typical to cherry pick test fixes. Is this cherry pick really needed?

SergeyKanzhelev commented:
I see a flake. Looks ok to merge

/approve
/lgtm
/triage accepted

claudiubelu commented:
/assign @derekwaynecarr

claudiubelu commented:
/assign @Random-Liu

claudiubelu committed:
tests: Updates the should delete a collection of pods test

We should wait for all 3 pods after we've spawned them, instead of
spawning and waiting for them sequentially.

claudiubelu commented:
/test pull-kubernetes-conformance-kind-ipv6-parallel

claudiubelu commented:
Release v1.21 reached EOL.
